k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -833,3 +932,48 @@ pub(crate) async fn get_source_version(ns: &str) -> error::Result<String> {
         .to_string();
     Ok(value.to_string())
 }
+
+/// Parse set-file and create config map data.
+pub(crate) async fn create_config_map_data(
+    upgrade_args: &UpgradeArgs,
+) -> error::Result<(BTreeMap<String, String>, HashMap<String, String>)> {
+    let mut data_map = BTreeMap::new();
+    let mut upgrade_map = HashMap::new();
+    let mut index = 1;
+    // for (index, file) in &upgrade_args.set_file .enumerate(){
+    for file in &upgrade_args.set_file {
+        let data: Vec<_> = file.split('=').collect();
+        let cm_values =
+            fs::read_to_string(data[1]).context(error::ReadFromFile { filepath: data[1] })?;
```

Comment:
this will panic if format incorrect..
suggest you do this at the top:
```rust
      let data: Vec<_> = file.split('=').collect();
      let [ _key, filepath ] = data[..] else {
          return Error .... not formatted properly;
      };
```

Reply:
You're still using data[1] rather than filepath, here and below